pykechain/models/property_reference.py

```diff
@@ -212,23 +216,45 @@ def _retrieve_objects(self, **kwargs) -> List[Context]:
 
 
 class StatusReferencesProperty(_ReferenceProperty):
-    """A virtual object representing a KE-chain Context References property.
+    """A virtual object representing a KE-chain Status References property.
 
     .. versionadded:: 3.19
     """
 
-    # REFERENCED_CLASS = Status
+    REFERENCED_CLASS = Status
 
-    def _retrieve_objects(self, **kwargs) -> List[Context]:
+    def _retrieve_objects(self, **kwargs) -> List[Status]:
         """
-        Retrieve a list of Contexts.
+        Retrieve a list of Statuses.
 
         :param kwargs: optional inputs
-        :return: list of Context objects
+        :return: list of Status objects
         """
         statuses = []
-        for statuse_json in self._value:
-            status = Status(client=self._client, json=statuse_json)
+        for status_json in self._value:
+            status = Status(client=self._client, json=status_json)
             status.refresh()  # To populate the object with all expected data
             statuses.append(status)
         return statuses
+
+
+class StoredFilesReferencesProperty(_ReferenceProperty):
+    """A virtual object representing a KE-chain StoredFile References property.
+
+    .. versionadded:: 4.7
+    """
+
+    REFERENCED_CLASS = StoredFile
+
+    def _retrieve_objects(self, **kwargs) -> List[StoredFile]:
+        """
+        Retrieve a list of StoredFile.
+
+        :param kwargs: optional inputs
+        :return: list of StoredFile objects
+        """
+        stored_files = []
+        for stored_files_json in self._value:
+            stored_file = StoredFile(client=self._client, json=stored_files_json)
+            stored_files.append(stored_file)
+        return stored_files
```

Comment:
Again nothing wrong with the code itself.
Out of pure curiosity. Have you considered using list comprehension if yes why you didn't use it?

```suggestion
        return [StoredFile(client=self._client, json=stored_files_json) for stored_files_json in self._value]
```

Reply:
Yeah, that's much neater. I was more doing it in a structured way, but this is better.